crates/ruma-state-res/src/event_auth.rs

```diff
@@ -128,22 +129,24 @@ pub fn auth_types_for_event(
 /// This assumes that `ruma_signatures::verify_event()` was called previously, as some authorization
 /// rules depend on the signatures being valid on the event.
 ///
-/// [authorization]: https://spec.matrix.org/latest/server-server-api/#authorization-rules
+/// # Errors
+///
+/// If the check fails, this returns an error with details about the check that failed.
```

Comment:
```suggestion
/// If the check fails, this returns an `Err(_)` with a description of the check that failed.
```

I think a `String` doesn't qualify for many definitions of "an error".